utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/client/CachedConfluentSchemaRegistryClient.scala

```diff
@@ -54,9 +55,27 @@ class CachedConfluentSchemaRegistryClient(val client: CSchemaRegistryClient, cac
 
     caches.schemaCache.getOrCreate(s"$subject-${schemaMetadata.getVersion}") {
       logger.debug(s"Cache parsed latest schema for subject: $subject, version: ${schemaMetadata.getVersion}.")
-      SchemaWithMetadata(schemaMetadata)
+      withExtraSchemaTypes(SchemaWithMetadata(schemaMetadata))
     }
   }
+
+  override def getSchemaById(id: Int): SchemaWithMetadata = {
+    //todo schemaCache is subject-version keyed, how to get it by id
+    val rawSchema = client.getSchemaById(id)
+    withExtraSchemaTypes(id, rawSchema)
+  }
+
+  private def withExtraSchemaTypes(id: Int, rawSchema: ParsedSchema) = {
+    (rawSchema.schemaType(), config.avroPlainTextSerialization) match {
+      case ("AVRO", Some(true)) => SchemaWithMetadata(AvroWithJsonPayloadSchema(rawSchema.asInstanceOf[AvroSchema]), id)
```

Comment:
`case (schema: AvroSchema, Some(true) => ` then, you wil get rid of casting in the following code